Documentation/building/freebsd-instructions.md

```diff
@@ -0,0 +1,278 @@
+# FreeBSD build instructions
+
+## Disclaimer
+Instructions bellow may or may not give you what you want.
+Tested on plain FreeBSD 11.3 Azure image
+
+## Prerequisites
+This needs to be done once on fresh system.
+
+```sudo pkg install cmake git icu libunwind bash python2 krb5 lttng-ust llvm60 libgit2```
+
+some scripts may still assume /bin/bash exists. To workaround it for now do of needed:
+```
+sudo ln -s /usr/local/bin/bash /bin/bash
+```
+
+This is certainly undesirable and it should be avoided if possible.
+
+# Get bootstrap cli
+```
+mkdir ~/dotnet
+cd ~/dotnet
+curl https://dotnetcli.blob.core.windows.net/dotnet/Sdk/master/dotnet-sdk-latest-freebsd-x64.tar.gz | tar xfz - 
+```
+if on 12.x you may also need to set `LD_PRELOAD` to `/usr/lib/libpthread.so` to avoid issue when cli freezes. 
+
+
+As of summer 2019 this CLI is no longer good enough to build all repos. If that is your case jump to section [Updating CLI](#updating--bootstrap-cli)
+Binary snapshot can be obtained from https://github.com/wfurt/blob as dotnet-sdk-freebsd-x64-latest.tgz
+
+## Getting sources
+master of source-build pulls in source code of specific snapshot instead of tip of master branches. 
```

Comment:
I'm having a hard time following these docs.  I'm going to stop commenting on it.

Reply:
Yes I see what you mean. The document needs to be rewritten.